consensus/spos/bls/blsSubroundsFactory.go

```diff
@@ -246,25 +273,53 @@ func (fct *factory) generateSignatureSubround() error {
 		fct.appStatusHandler,
 	)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	subroundSignatureObject, err := NewSubroundSignature(
+	subroundSignatureInstance, err := NewSubroundSignature(
 		subround,
 		fct.worker.Extend,
 		fct.appStatusHandler,
 	)
+	if err != nil {
+		return nil, err
+	}
+
+	return subroundSignatureInstance, nil
+}
+
+func (fct *factory) generateEndRoundSubround() error {
+	subroundEndRoundInstance, err := fct.generateEndRoundSubroundV1()
 	if err != nil {
 		return err
 	}
 
-	fct.worker.AddReceivedMessageCall(MtSignature, subroundSignatureObject.receivedSignature)
-	fct.consensusCore.Chronology().AddSubround(subroundSignatureObject)
+	switch fct.consensusModel {
```

Comment:
Can be made in another PR: instead of having multiple `switch` operations on each subround, we can have only one switch to create all the subrounds following the model 1 or model 2.